azure-arc-for-servers/grant-permission.ps1

```diff
@@ -0,0 +1,43 @@
+# you can run anywhere
+
+function Get-AzAccountAccessToken()
+{
+    if(-not (Get-Module Az.Accounts)) {
+        Import-Module Az
+    }
+
+    $audience ="https://management.core.windows.net/"       
+    $context = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureRmProfileProvider]::Instance.Profile.DefaultContext
+    $token = [Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate($context.Account, $context.Environment, $context.Tenant.Id.ToString(), $null, [Microsoft.Azure.Commands.Common.Authentication.ShowDialog]::Never, $null, $audience).AccessToken
+    return $token
+}
+
+function Set-KeyVaultAccessPolicy {
+    [CmdletBinding()]
+    param (
+       [Parameter(Mandatory=$true)]
+       [string] $machineName,
+       [Parameter(Mandatory=$true)]
+       [string] $subscriptionId,
+       [Parameter(Mandatory=$true)]
+       [string] $resourceGroup,
+       [Parameter(Mandatory=$true)]
+       [string] $keyVaultName,
+       [string] $apiVersion ="2019-08-15"
+    )
+    
+    $machineUri= "https://management.azure.com/subscriptions/${subscriptionId}/resourcegroups/${resourceGroup}/providers/Microsoft.HybridCompute/machines/${machineName}?api-version=${apiVersion}"
```

Comment:
What is the purpose of the braces {} in `$machineUri`? They are not required.

Reply:
I think it's for "escaping" the variable name. Without ${}, sometimes fails

Reply:
Those variable names don't need escaping there. The only one that needs escaping it $machinename because of the `?` is seen as a valid character in a variable name. So the braces force the termination of the variable name for the parser.

```powershell
$machineUri= "https://management.azure.com/subscriptions/$subscriptionId/resourcegroups/$resourceGroup/providers/Microsoft.HybridCompute/machines/${machineName}?api-version=$apiVersion"

# or 

$machineUri= "https://management.azure.com/subscriptions/$subscriptionId/resourcegroups/$resourceGroup/providers/Microsoft.HybridCompute/machines/$machineName" + "?api-version=$apiVersion"
```